### PR TITLE
Feature/371 single tenant deployment

### LINKED ISSUE #371: Support single-tenant deployments
We foresee the likelihood of countries wanting to have their own server in-country. A CDx server instance should be able to be configured as 'single-tenant', meaning basically that users cannot self-register. Multiple institutions can be created, but they will have to be created by the server super-admin

### PR DESCRIPTION
**Pull request by nekron**
Fixes #371.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -30,9 +30,15 @@ def create(model)
   end
 
   def implicit_policies
-    self.institutions.pluck(:id, :kind).map do |institution_id, kind|
+    policies = self.institutions.pluck(:id, :kind).map do |institution_id, kind|
       Policy.owner(self, institution_id, kind)
-    end + [Policy.implicit(self)].compact
```
Comment (spalladino) [inline app/models/user.rb]: This change can be implemented in this block:
```ruby
self.institutions.pluck(:id, :kind).map do |institution_id, kind|
  Policy.owner(self, institution_id, kind)
end + [Policy.implicit(self) unless Settings.single_tenant].compact
```

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -61,19 +54,23 @@ def authorize_resource(resource, action)
   def check_no_institution!
     return if current_user && current_user.need_change_password?
     if current_user && current_user.institutions.empty? && current_user.policies.empty?
-      redirect_to new_institution_path
+      if has_access?(Institution, CREATE_INSTITUTION)
+        redirect_to new_institution_path
+      else
+        @hide_nav_bar = true
```
Comment (spalladino) [inline app/controllers/application_controller.rb]: Should the user be redirected somewhere else in this case?

Comment (nekron) [inline app/controllers/application_controller.rb]: Eventually, when the user has the ability to request access to an institution he will be redirected to that page.

Comment (nekron) [inline app/controllers/application_controller.rb]: I'll create a "contact your administrator" page for now.

**Pull request merged**